src/app/shared/components/xterm-terminal/component.ts

```diff
@@ -0,0 +1,215 @@
+    this.clusterId = this._activatedRoute.snapshot.paramMap.get(PathParam.ClusterID);
+
+    // Note:
+    // Get Cluster Info inorder to show cluster name in shell
+    if (this.projectId && this.clusterId) {
+      this._clusterService
+        .cluster(this.projectId, this.clusterId)
+        .pipe(take(1))
+        .subscribe({
+          next: (cluster: Cluster) => {
+            this.cluster = cluster;
+            this.initTerminal();
+          },
+          error: _ => {},
+        });
+    }
+
+    // Subscription bidirectional communication
+    this.websocketService.messages$.pipe(takeUntil(this._unsubscribe)).subscribe((data: ITerminalFrame) => {
+      this.handleWebSocketConnectionMessages(data);
+    });
+
+    // Subscription connection success
+    this.websocketService
+      .getWebSocketOnConnectObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        this._notificationService.success('Successfully connected to cloud shell terminal');
+        this.isConnectionLost = false;
+      });
+
+    // Subscription connection lost
+    this.websocketService
+      .getWebSocketOnCloseObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        this.isConnectionLost = true;
+      });
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+    this.websocketService.close();
+
+    // Clean 'resize' Event listener subscription
+    this.removeEventListener();
+  }
+
+  ngAfterViewInit(): void {
+    this.connectToWebSocketConnection();
+  }
+
+  initTerminal(): void {
+    if (this.terminal) {
+      this.terminal.dispose();
+    }
+
+    this.terminal = new Terminal({
+      fontSize: 14,
+      fontFamily: 'Consolas, "Courier New", monospace',
+      bellStyle: 'sound',
+      cursorBlink: true,
+      theme: {
+        foreground: '#f8f8f8',
+        background: '#2b3035',
+      },
+    });
+
+    const containerElement = this.terminalRef.nativeElement;
+    const fitAddon = new FitAddon();
+    this.terminal.loadAddon(fitAddon);
+    this.terminal.open(containerElement);
+
+    const clusterName = this.cluster && this.cluster.name;
+    this.terminal.write('\x1b[37mWEB TERMINAL\x1b\r\n\n');
+    this.terminal.write('\x1b[37mWelcome to Cloud Shell! Type "help" to get started.\r\n\n');
+    this.terminal.write(
+      `\x1b[37mYour Cloud Platform project in this session is set to \x1b[1;34m${clusterName}\x1B[0m\n\n\r`
+    );
+
+    const delayFn = debounce(() => {
+      fitAddon.fit();
+    }, this.DELAY_TIMEOUT);
+    delayFn();
+
+    this.removeEventListener();
+    // Event listeners binding
+    this._resizeEventlistenerFn = this._renderer.listen(window, 'resize', () => {
+      delayFn();
+    });
+
+    // XTerminal Properties
+    this.terminal.onData(this.onTerminalSendString.bind(this));
+  }
+
+  onCloseTerminal(): void {
+    this.terminalDisposed.emit();
+  }
+
+  onReconnect() {
```

Comment:
Return type is missing here and in many other methods.

Reply:
@KhizerRehan this is not resolved, check all your changes.

Reply:
Fixed commit [`6e41851` (#4492)](https://github.com/kubermatic/dashboard/pull/4492/commits/6e41851d63882950dd1b1a529f0d08a56556364c)

Reply:
I've added comments in places where it is still not done, i.e. https://github.com/kubermatic/dashboard/pull/4492/files#diff-cd8604576285d61d53d3648ae03e91dbe49e58aa6f262733dccadfbc79316dd7R531.